File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.java
class Solution {
    public int smallestNumber(int n, int t) {
        int p = 1;
        char[] num = Integer.toString(n).toCharArray();
        for(char ch : num){
            p *= (ch - '0');
        }
        if(p % t == 0)  return n;
        return smallestNumber(n+1,t);
    }
}

File: 3345-smallest-divisible-digit-product-i/3345-smallest-divisible-digit-product-i.kt
class Solution {
    fun smallestNumber(n: Int, t: Int): Int {
        var num = n.toString().toCharArray()
        var p = 1
        for(ch in num){
            p *= (ch - '0')
        }
        if(p % t == 0)  return n;
        return smallestNumber(n+1,t)
    }
}